edgepi-leds.html

```diff
@@ -1,74 +1,89 @@
 <script type="text/javascript">
-    RED.nodes.registerType('led-trigger-node',{
+    RED.nodes.registerType('edgepi-led-node',{
         category: 'Osensa EdgePi',
         color: '#a6bbcf',
         defaults: {
             name: { value:"" },
+            ConfigStyle: { value: "Editor" },
+            Method: { value: "toggleLed"},
+            LedPin: { value: "LED1" }
         },
         inputs:1,
         outputs:1,
         icon: "light.svg",
         label: function() {
-            return this.name||"led-trigger-node";
+            return this.name||"edgepi-led-node";
+        },
+        oneditprepare: function(){
+            // Query html elements
+            const configStyleSelect = document.getElementById("node-input-ConfigStyle");
+            const methodRow = document.querySelector(".form-row.Method");
+            const ledPinRow = document.querySelector(".form-row.LEDPin");
+            const receiveInputInfo = document.querySelector(".ReceiveInput-INFO-Container")
+
+            function updateEditor(){
+                if (configStyleSelect.value === "Editor") {
+                    methodRow.style.display = "block";
+                    ledPinRow.style.display = "block";
+                    receiveInputInfo.style.display = "none";
+                } else {
+                    methodRow.style.display = "none";
+                    ledPinRow.style.display = "none";
+                    receiveInputInfo.style.display = "block"
+                }
+            }
+            // Update and add event listener to update on user input
+            updateEditor()
+            configStyleSelect.addEventListener("change", updateEditor);
         }
     });
 </script>
```

Comment:
why is this javascript stuff in an html file? isn't it better to separate them?

Reply:
Yes it's usually better to separate them. However in the case of node-red, this how they require it to be done. I'm not sure how it works under the hood on the node-red side, but has something to do with the way they parse the html file and generate the node UI.